File: src/main/frontend/src/app/core/time-service.ts
import { Injectable } from '@angular/core';

/**
 * Facade over the browser wall clock. All client-side time reads must go through this service
 * (never `Date.now()` / `new Date()` directly) so tests can substitute a fixed clock — via a
 * `{ provide: TimeService, useValue: … }` override — and assert against exact, repeatable
 * values.
 *
 * The app currently has no direct time read (server timestamps are rendered as-is); this is the
 * sanctioned entry point for any future one.
 */
@Injectable({ providedIn: 'root' })
export class TimeService {
  /** Current epoch milliseconds (replaces `Date.now()`). */
  now(): number {
    return Date.now();
  }

  /** Current date/time (replaces `new Date()`). */
  nowDate(): Date {
    return new Date();
  }
}

File: src/main/java/tech/dobler/werstreamt/time/TimeService.java
package tech.dobler.werstreamt.time;

import java.time.Instant;
import java.time.LocalDate;

/**
 * Facade over the JVM wall clock. Production code reads "now" exclusively through this service
 * (never {@link Instant#now()} / {@link LocalDate#now()} directly) so that tests can substitute
 * a fixed clock and assert against exact, repeatable timestamps.
 */
public interface TimeService {

    /** The current instant (replaces {@link Instant#now()}). */
    Instant now();

    /** Today's date in the system zone (replaces {@link LocalDate#now()}). */
    LocalDate today();
}
